### PR TITLE
feat: [CHAOS-171]: Supporting inline manifest

### PR DESCRIPTION
**Pull request by hextechpal**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35602)

**Comment by hextechpal** [comment]
trigger messagecheck

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -177,8 +177,7 @@ templateServiceClientConfig:
   readTimeOutSeconds: 15
 
 chaosServiceClientConfig:
-  # todo confirm endpoint
-  baseUrl: http://localhost:15001/api/
+  baseUrl: http://localhost:8080
```
Comment (imrajdas) [inline 120-ng-manager/config.yml:180]: Port should be 9000, callback function has been moved to chaos-driver

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)* status: 16 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosNotifyCallback.java` line 48 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8gyfzd1CBeW8jsXXGr:-N8gyfzd1CBeW8jsXXGs:b-5z1tx4) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosNotifyCallback.java#L48)):*
> ```Java
>     }
> 
>     Boolean isSuccessful = NGRestUtils.getResponse(chaosHttpClient.pushTaskResponse(responseBuilder.build()));
> ```

it might throw exception in case of not being successful 
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 44 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8h-2do9ecgHCrQLuB7:-N8h-2do9ecgHCrQLuB8:bd7wc55) ([raw file](https://github.com/harness/harness-core/blob/0f88473f957070af899510ad5f29b0e867844342/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L44)):*
> ```Java
>   }
> 
>   private TaskParameters getTaskParams(ChaosK8sRequest request) {
> ```

why public?
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 51 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8gyv543C3jQwtAIHNr:-N8gyv543C3jQwtAIHNs:bikq729) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L51)):*
> ```Java
>     LinkedHashMap<String, String> logAbstractions = new LinkedHashMap<>();
>     logAbstractions.put("accountId", chaosK8sRequest.getAccountId());
>     logAbstractions.put("chaosUid", chaosUid);
> ```

chaosUuid 

Also make these as constants
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 56 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8gz0oM6-9lIOBsECU0:-N8gz0oM6-9lIOBsECU1:b-aw9oe5) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L56)):*
> ```Java
>             .accountId(chaosK8sRequest.getAccountId())
>             .taskParameters(getTaskParams(
>                 chaosK8sRequest.getAccountId(), chaosK8sRequest.getK8sConnectorId(), chaosK8sRequest.getK8sManifest()))
> ```

lets give whole chaosk8srequest object to lower layer and let it use what it wants as in future we can have more params
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 88 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8h-6XJ2U6JkVQMQ8Oy:-N8h-6XJ2U6JkVQMQ8Oz:bwldkf4) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L88)):*
> ```Java
>         .skipDryRun(true)
>         .releaseName("chaos-" + generateUuid())
>         .commandName("K8s Apply")
> ```

K8sApply Enum .getname?
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/client/ChaosApplyManifestResponse.java` line 10 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8h-FZ3AyxpCxXL3yKW:-N8h-FZ3AyxpCxXL3yKX:b-88n4iv) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/client/ChaosApplyManifestResponse.java#L10)):*
> ```Java
> public class ChaosApplyManifestResponse {
>   String taskId;
>   String status; // SUCCESS FAILED
> ```

Enum?

**Review by hextechpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)* status: 16 file(s) to review. 1 change requests. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/config.yml` line 180 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8cHP36kY_Liq0IrlTB-r5-180:-N8w6ShC5X9SOjGxQwLR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/config.yml#L180)):*
<details><summary><i>Previously, imrajdas (Raj Das) wrote…</i></summary><blockquote>

Port should be 9000, callback function has been moved to chaos-driver
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosNotifyCallback.java` line 48 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8gyfzd1CBeW8jsXXGr:-N8w6USl8A33gXTlevzh:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosNotifyCallback.java#L48)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

it might throw exception in case of not being successful 
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 44 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8h-2do9ecgHCrQLuB7:-N8w70dbCTHFPiKb0zVf:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0f88473f957070af899510ad5f29b0e867844342/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L44)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

why public?
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 51 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8gyv543C3jQwtAIHNr:-N8w6W96C5iQ4UGX3aGC:b-g903i8) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L51)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

chaosUuid 

Also make these as constants
</blockquote></details>

chaosUuid is something that will help us tract this request across wait engine.

No point of constants here have extracted this to a method though
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 56 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8gz0oM6-9lIOBsECU0:-N8w6r2K6_qDzBZAB947:b7gg70g) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L56)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

lets give whole chaosk8srequest object to lower layer and let it use what it wants as in future we can have more params
</blockquote></details>

I think everything in the request will be needed here no point bloating the signatures. If required we can create a args object later
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java` line 88 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8h-6XJ2U6JkVQMQ8Oy:-N8w71YlCN6gAzESX711:b33wjfd) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/ChaosServiceImpl.java#L88)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

K8sApply Enum .getname?
</blockquote></details>

no enum extracted in a constant
___
*[`120-ng-manager/src/main/java/io/harness/ng/chaos/client/ChaosApplyManifestResponse.java` line 10 at r5](https://reviewable.io/reviews/harness/harness-core/35602#-N8h-FZ3AyxpCxXL3yKW:-N8w74VrE8G_FFnoXHpJ:b4xg1y6) ([raw file](https://github.com/harness/harness-core/blob/9d9d60a46281a85f2da8919c58a2bb6162f2bbf8/120-ng-manager/src/main/java/io/harness/ng/chaos/client/ChaosApplyManifestResponse.java#L10)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

Enum?
</blockquote></details>

will do later if required

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 2 of 10 files at r1, 3 of 5 files at r2, 7 of 11 files at r4, 1 of 2 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)* status: 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602#-:-N8wVSU13a_VAL_o0SWI:bnfp4nl)* status: 1. unresolved discussions.

**Review by imrajdas — COMMENTED** [review]
> Reviewed 2 of 10 files at r1, 7 of 11 files at r4, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)* status: 1. unresolved discussions.

**Review by imrajdas — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)* status: 1. unresolved discussions.

**Review by imrajdas — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35602)* status: :shipit: complete! ✓

**Pull request merged**